src/components/common/checkbox/CheckBox.tsx

```diff
@@ -0,0 +1,71 @@
+import clsx from 'clsx';
+import { forwardRef } from 'react';
+import type { MouseEvent } from 'react';
+
+import { CheckBoxProps } from './CheckBox.types';
+import { CheckBoxIcon } from './CheckBoxIcon';
+
+import Label from '@/components/common/label/Label';
+
+const CheckBox = forwardRef<HTMLInputElement, CheckBoxProps>(function CheckBox(
+  { id, isIndeterminate, checked, disabled, className, labelText, ...restProps },
+  ref,
+) {
+  const baseStyles =
+    'group relative outline-none radius-4xs before:absolute duration-faster ease-(--motion-fluent) before:inset-0 before:rounded-[inherit] inline-flex items-center justify-center border';
+
+  const enabledStyles =
+    checked || isIndeterminate
+      ? 'bg-feedback-information-dark border-border-trans-normal-dark'
+      : 'bg-object-static-inverse-hero-dark border-border-normal-dark';
+
+  const disabledStyles =
+    checked || isIndeterminate
+      ? 'bg-feedback-trans-information-dark border-border-trans-assistive-dark'
+      : 'bg-object-static-inverse-disabled-dark border-border-trans-assistive-dark';
+
+  const handleClick = (e: MouseEvent<HTMLInputElement>) => {
+    e.currentTarget.blur();
+    if (restProps.onClick) {
+      restProps.onClick(e);
+    }
+  };
+
+  return (
+    <div className={clsx('gap-2xs inline-flex items-center', className)}>
+      <div
+        className={clsx(
+          baseStyles,
+          disabled ? disabledStyles : enabledStyles,
+          'focus-within:before:shadow-focus-visible group-hover:before:bg-[rgba(26,27,35,0.12)] group-active:before:bg-[rgba(26,27,35,0.12)]',
+        )}
+      >
```

Comment:
```suggestion
      <div
        tabIndex={0}
        className={clsx(
          baseStyles,
          disabled ? disabledStyles : enabledStyles,
          'interaction-default-bold-inverse active:shadow-focus-visible duration-faster ease-(--motion-fluent)',
        )}
      >
```

직접 해보니까 기존 디자인 토큰을 사용해서 충분히 구현 가능하더라구요
group을 사용하지 않아도 동작이 되었는데 제 생각에는 input이 hover되었을 때 버블링이 일어나서 부모요소 hover가 트리거되는 것 같습니다!

> checkbox 클릭 시 focus가 자동으로 되는데, 이 focus는 다른 창이나 외부 요소를 클릭 전까지 유지됩니다. 레퍼런스 처럼 클릭 시 focus가 작동하고 즉시 해제되는 플로우를 위해 반영했습니다.

focus는 요소가 포커스되면 다른 요소로 포커스 되기 전까지 효과를 유지하는 의사 클래스입니다. 대신에 요소에 마우스 커서를 올려 놓고 누르는 순간부터 떼는 시점까지 해당 요소를 선택하는 `active`를 사용하면 클릭하는 시점에서만 동작하도록 구현할 수 있습니다!

Reply:
이렇게 하면 handleClick도 없어도 잘 나올 거예요

Reply:
++) active에는 before 가상요소를 붙여주지 않은 이유는 체크박스 자체에 border가 있기 때문입니다,..!  border도 같이 나와야하기때문에 #87 에서 설명한 것처럼 before가 아닌 선택한 요소 자체에 걸어주었습니다!

Reply:
원인을 다시 확인해봤는데, TabIndex 문제였네요. 웬만하면 해당 요소를 사용하지 않고, 실제 태그 위주로 사용하려고 하는데, 이 경우에는 <Icon>의 구현 방식과 <input> 의 opacity 값 때문에 해당 방식으로 수정하였습니다~

Reply:
`tabIndex={0}` 가 존재하지 않아도 기존 Interaction 자체에는 문제가 없고, click 시에 focus가 들어가는 방식에는 전혀 문제가 없는데, 실제 tab으로 focus 이동 시에 문제가 생길 수 있다고 판단하였습니다!

Reply:
[8692706](https://github.com/JECT-Study/JECT-Official-WebSite-Client/pull/86/commits/869270618bf7d57c4171274c11d66ff063ae6bb7) 에서 수정했습니다!

Reply:
저도 tabIndex 지정보단 탭 이동이 가능한 태그를 사용하는 쪽이 좋은 방향이라고 생각해요! 
input 태그가 잡히도록 하면 베스트이겠지만 쉽지 않네요 div태그를 button 태그로 변경하는 방법도 있긴합니다!

++) 수정 확인해보니 아직 tab 이동은 불가해보입니다!

Reply:
[7dd6763](https://github.com/JECT-Study/JECT-Official-WebSite-Client/pull/86/commits/7dd67639491fdd46a1a8814cb68ef65b74592adc) 에서 tabIndex 부여했습니다!!